node/src/components/linear_chain.rs

```diff
@@ -20,14 +23,46 @@ use crate::{
         EffectBuilder, EffectExt, EffectResultExt, Effects,
     },
     protocol::Message,
-    types::{BlockByHeight, FinalitySignature, Timestamp},
+    types::{BlockByHeight, BlockSignatures, FinalitySignature, Timestamp},
     NodeRng,
 };
+use casper_types::{EraId, ProtocolVersion};
 
 pub use event::Event;
-pub(crate) use state::LinearChain;
+use state::LinearChain;
 
-impl<I, REv> Component<REv> for LinearChain<I>
+#[derive(DataSize, Debug)]
+pub(crate) struct LinearChainComponent<I> {
+    linear_chain_state: LinearChain,
+    #[data_size(skip)]
+    metrics: LinearChainMetrics,
+    _marker: PhantomData<I>,
```

Comment:
Let's get rid of this `PhatomData` marker field

Reply:
That wouldn't compile, would it?